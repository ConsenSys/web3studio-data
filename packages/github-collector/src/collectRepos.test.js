const fetchMock = require('node-fetch').default;

const { toArray, map } = require('rxjs/operators');
const collectRepos = require('./collectRepos');
const orgReposResponse = require('../test/fixtures/organizationRepositories');

describe('collectRepos', () => {
  beforeAll(() => {
    fetchMock.post('*', () => orgReposResponse());
  });

  afterAll(() => {
    fetchMock.reset();
  });

  it("filters repos that don't have web3studio project topics", async () => {
    const results = await collectRepos()
      .pipe(toArray())
      .toPromise();

    expect(results.length).toEqual(11);

    results.forEach(repo =>
      repo.projects.forEach(project =>
        expect(project).toEqual(expect.stringMatching(/^web3studio-/))
      )
    );
  });

  it('returns the name and owner of repos', async () => {
    const results = await collectRepos()
      .pipe(
        map(repo => ({ name: repo.name, owner: repo.owner })),
        toArray()
      )
      .toPromise();

    expect(results).toMatchSnapshot();
  });

  it('has a list of projects based on topics', async () => {
    const projects = await collectRepos()
      .pipe(
        map(repo => repo.projects),
        toArray()
      )
      .toPromise();

    expect(projects).toMatchSnapshot();
  });
});
