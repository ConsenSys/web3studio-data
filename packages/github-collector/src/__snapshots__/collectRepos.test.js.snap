// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collectRepos has a list of projects based on topics 1`] = `
Array [
  Array [
    "web3studio-internal",
  ],
  Array [
    "web3studio-internal",
  ],
  Array [
    "web3studio-soy",
  ],
  Array [
    "web3studio-sojourn",
  ],
  Array [
    "web3studio-projects",
  ],
  Array [
    "web3studio-internal",
  ],
  Array [
    "web3studio-internal",
  ],
  Array [
    "web3studio-bootleg",
  ],
  Array [
    "web3studio-blackholesum",
  ],
  Array [
    "web3studio-internal",
  ],
  Array [
    "web3studio-sojourn",
  ],
]
`;

exports[`collectRepos returns the name and owner of repos 1`] = `
Array [
  Object {
    "name": "web3studio-website",
    "owner": "ConsenSys",
  },
  Object {
    "name": "web3studio-standards",
    "owner": "ConsenSys",
  },
  Object {
    "name": "web3studio-soy",
    "owner": "ConsenSys",
  },
  Object {
    "name": "web3studio-sojourn",
    "owner": "ConsenSys",
  },
  Object {
    "name": "web3studio-projects",
    "owner": "ConsenSys",
  },
  Object {
    "name": "web3studio-meshgame",
    "owner": "ConsenSys",
  },
  Object {
    "name": "web3studio-data",
    "owner": "ConsenSys",
  },
  Object {
    "name": "web3studio-bootleg",
    "owner": "ConsenSys",
  },
  Object {
    "name": "web3studio-blackHoleSum",
    "owner": "ConsenSys",
  },
  Object {
    "name": "web3studio",
    "owner": "ConsenSys",
  },
  Object {
    "name": "react-native-sssa",
    "owner": "ConsenSys",
  },
]
`;
